### PR TITLE
added new nginx streaming guide

### LINKED ISSUE #3: Update nginx guide

### PR DESCRIPTION
**Pull request by mrbar42**
fixes #3

### CODE REVIEW

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
+## Setting up HLS live streaming server using NGINX + nginx-rtmp-module on Ubuntu
+
+### Stages:
+
+1. compile nginx with rtmp module
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: link each headline to the matching chapter

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+3. start nginx
+4. push the live stream using rtmp
+5. take the server for a test run!
+
+
+## 1. Compile nginx with rtmp module
+
+Firstly, we'll need to compile nginx with the nginx-rtmp-module.
+
+Even so that the original module can be found [here](https://github.com/arut/nginx-rtmp-module), there is a [fork](https://github.com/sergey-dryabzhinsky/nginx-rtmp-module), that continues the development and adds fixes and improvements actively.
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: I'd rephrase:
We recommend using [this] forked module, it's being actively worked on and contains more fixes and improvements over the [original one].

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+## 1. Compile nginx with rtmp module
+
+Firstly, we'll need to compile nginx with the nginx-rtmp-module.
+
+Even so that the original module can be found [here](https://github.com/arut/nginx-rtmp-module), there is a [fork](https://github.com/sergey-dryabzhinsky/nginx-rtmp-module), that continues the development and adds fixes and improvements actively.  
+ there for the fork is the recommended one.
+
+Clone the module to a local directory
+
+    git clone https://github.com/sergey-dryabzhinsky/nginx-rtmp-module.git
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: put code inside code snippets
```
code
```

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    sudo apt-get install build-essential libpcre3 libpcre3-dev libssl-dev
+
+latest nginx can be downloaded from [this page](http://nginx.org/en/download.html).  
+for example `nginx-1.10.1` can be downloaded from this link: [http://nginx.org/download/nginx-1.10.1.tar.gz](http://nginx.org/download/nginx-1.10.1.tar.gz) 
+  
+    wget http://nginx.org/download/nginx-1.10.1.tar.gz
+    tar -xf nginx-1.10.1.tar.gz
+    cd nginx-1.10.1
+  
+notice the `--add-module=../nginx-rtmp-module` argument, the path must point correctly to the cloned module
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: notice statement should come after what it specifies to notice.
Also the styling looks abit cramped together, the configuration and the download are too close together IMO

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+        }
+    }
+}
+```
+
+Note that the example points `/mnt/hls/` as the target path.  
+change this to a different directory but make sure that nginx have **write permissions**.  
+to fix permissions you can use this command:
+
+    chown -R nginx:nginx /mnt/hls
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: when installing nginx it sets a user/group named nginx?

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+            video/mp2t ts;
+        }
+
+        root /mnt/;
+    }
+}
+```
+
+
+The default location for nginx conf is `/usr/local/nginx/conf/nginx.conf` or `/etc/nginx/nginx.conf`
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: note the location at the start of this chapter

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+  
+-  `-re` - consume stream on media's native bitrate (and not as fast as possible)
+-  `-f` - use video4linux2 plugin
+-  `-i` - select physical device to capture from
+-  `-vcodec` - specify video codec to output
+-  `-vprofile` - use x264 baseline profile
+-  `-acodec` - use aac audio codec
+-  `-strict` - allow using the experimental aac codec
+-  `-f` - specify format to output
+-  `rtmp://localhost/show/stream` - rtmp endpoint to stream to. if the target port is not `1935` is should be included in the uri.
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: add that in our case it's the rtmp location.
Also how come there's /stream is that what nginx expect in addtion to the location?

File: docs/Guides/Setting up HLS live streaming server using NGINX.md [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+<script>
+    var player = videojs('#player');
+</script>
+</body>
+</html>
+```
+
+
+with Peer5 plugin
+get your API key [here](https://app.peer5.com)
```
Comment (shacharz) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: link to app.peer5.com/register

**Comment by shacharz** [comment]
Let's add that we recommend 20 segments of 3 seconds each, and add that configuration

**Comment by shacharz** [comment]
@whadar Please review as well to see if you're happy with it

File: docs/Guides/Setting up HLS live streaming server using NGINX.md (hunk 1 above)
Comment (mrbar42) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: The link was removed completely because they exist on on mkdocs's menu

File: docs/Guides/Setting up HLS live streaming server using NGINX.md (hunk 5 above)
Comment (mrbar42) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: good point... including nginx user reference is too long. i'm omitting the command entirely

File: docs/Guides/Setting up HLS live streaming server using NGINX.md (hunk 6 above)
Comment (mrbar42) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: after compiling nginx and running `make install` nginx will be installed in `/usr/local/nginx/`   
but i also included `/etc/nginx` because its the default path of non-manual installation.

the `/mnt/hls`  relates to the path of the hls files (also added that)

File: docs/Guides/Setting up HLS live streaming server using NGINX.md (hunk 7 above)
Comment (mrbar42) [inline docs/Guides/Setting up HLS live streaming server using NGINX.md]: added explination before in the nginx section
